src/twitter.rs

```diff
@@ -38,77 +53,122 @@ pub async fn supervisor(
 
     loop {
         futures::select! {
+            // We were scheduled to restart, we verify that anyone is subscribed
+            // and start a new stream
             () = restart => {
-                let follows : Vec<u64> = requested_follows.union(&default_follows).copied().collect();
-                if !follows.is_empty() {
-                    twitter_stream.set(stream_consumer(config.token(), follows, tx_tweet.clone()).fuse());
-
-                    backing_off = false;
-                } else {
-                    log::warn!("not starting stream, nothing to follow yet");
-                    restart.set(delay_for(Duration::from_secs(5)).fuse());
+                backing_off = false;
+
+                if requested_follows.is_empty() {
+                    if twitter_stream.is_terminated().not() {
+                        log::warn!("closing existing stream");
+                        twitter_stream.set(Fuse::terminated());
+                    }
+
+                    log::info!("no follows were requested, let's wait some more");
+                    continue;
                 }
+
+                let mut follows = Follows::new();
+                follows.extend(requested_follows.keys());
+                follows.extend(&config.default_follows);
```

Comment:
default_follows are never considered in logic, begging the question whether it should be kept or not